staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go

```diff
@@ -219,94 +213,20 @@ func TestWatch(t *testing.T) {
-	cacher, _, err := newTestCacherWithClock(etcdStorage, fakeClock)
-	if err != nil {
-		t.Fatalf("Couldn't create cacher: %v", err)
-	}
-	defer cacher.Stop()
-
-	podFoo := makeTestPod("foo")
-	podBar := makeTestPod("bar")
-
-	podFooPrime := makeTestPod("foo")
-	podFooPrime.Spec.NodeName = "fakeNode"
-
-	podFooBis := makeTestPod("foo")
-	podFooBis.Spec.NodeName = "anotherFakeNode"
-
-	podFooNS2 := makeTestPod("foo")
-	podFooNS2.Namespace += "2"
-
-	// initialVersion is used to initate the watcher at the beginning of the world,
-	// which is not defined precisely in etcd.
-	initialVersion, err := cacher.LastSyncResourceVersion()
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	startVersion := strconv.Itoa(int(initialVersion))
-
-	// Set up Watch for object "podFoo".
-	watcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: startVersion, Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer watcher.Stop()
-
-	// Create in another namespace first to make sure events from other namespaces don't get delivered
-	updatePod(t, etcdStorage, podFooNS2, nil)
-
-	fooCreated := updatePod(t, etcdStorage, podFoo, nil)
-	_ = updatePod(t, etcdStorage, podBar, nil)
-	fooUpdated := updatePod(t, etcdStorage, podFooPrime, fooCreated)
-
-	verifyWatchEvent(t, watcher, watch.Added, podFoo)
-	verifyWatchEvent(t, watcher, watch.Modified, podFooPrime)
-
-	initialWatcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: fooCreated.ResourceVersion, Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer initialWatcher.Stop()
-
-	verifyWatchEvent(t, initialWatcher, watch.Modified, podFooPrime)
-
-	// Now test watch from "now".
-	nowWatcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: "0", Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
-	defer nowWatcher.Stop()
-
-	verifyWatchEvent(t, nowWatcher, watch.Added, podFooPrime)
-
-	_ = updatePod(t, etcdStorage, podFooBis, fooUpdated)
-
-	verifyWatchEvent(t, nowWatcher, watch.Modified, podFooBis)
-
-	// Add watchCacheDefaultCapacity events to make current watch cache full.
-	// Make start and last event duration exceed eventFreshDuration(current 75s) to ensure watch cache won't expand.
-	for i := 0; i < watchCacheDefaultCapacity; i++ {
-		fakeClock.SetTime(time.Now().Add(time.Duration(i) * time.Minute))
-		podFoo := makeTestPod(fmt.Sprintf("foo-%d", i))
-		updatePod(t, etcdStorage, podFoo, nil)
-	}
-
-	// Check whether we get too-old error via the watch channel
-	tooOldWatcher, err := cacher.Watch(context.TODO(), "pods/ns/foo", storage.ListOptions{ResourceVersion: "1", Predicate: storage.Everything})
-	if err != nil {
-		t.Fatalf("Expected no direct error, got %v", err)
-	}
-	defer tooOldWatcher.Stop()
+func TestWatchFromZero(t *testing.T) {
```

Comment:
BTW - it might make sense to split this PR into two and having this part merged earlier - may require less work to make it happen and already provides some benefits.